src/filter/time.rs

```diff
@@ -58,107 +67,96 @@ mod tests {
     use super::*;
     use std::time::Duration;
 
+    fn set_test_time(time: Zoned) -> SystemTime {
```

Comment:
To avoid leaking state between tests, I think an API like this would be better:

```rust
fn with_test_time(time: Zoned, f: impl FnOnce()) -> SystemTime
```

And have it reset TESTTIME to `None` with a `Drop` handler.

Reply:
... or, since you need a drop handler anyway, maybe just

```rust
struct TestTimeGuard;

impl TestTimeGuard {
    fn new(time: Zoned) -> Self { ... }
}

impl Drop for TestTimeGuard { ... }
```